src/wp-includes/class-wp-widget.php

```diff
@@ -612,12 +612,16 @@ public function get_settings() {
 		$settings = get_option( $this->option_name );
 
 		if ( false === $settings ) {
+			$settings = array();
 			if ( isset( $this->alt_option_name ) ) {
-				$settings = get_option( $this->alt_option_name );
-			} else {
-				// Save an option so it can be autoloaded next time.
-				$this->save_settings( array() );
+				// Get from legacy (alternative) option name.
+				$settings = get_option( $this->alt_option_name, array() );
+
+				// Delete the legacy option as the new option will be created.
```

Comment:
Do we need to describe here which new option will be created?

Reply:
Such as:
> // Delete the legacy option as the new option will be created using `$this->option_name`.